### ISSUE TITLE
Custom analyzer is not applied for wildcard queries

### ISSUE THREAD

**Issue by andreybratukhin** [elastic/elasticsearch#22540]
**Elasticsearch version**: 5.1.1

**Steps to reproduce**:
 It seems this change (https://github.com/elastic/elasticsearch/pull/21919) introduces inconsistent behavior for prefix and wildcard queries. Here is example of my settings:

```
{
  "settings": {
    "analysis": {
      "analyzer": {
        "default": {
          "tokenizer": "whitespace",
          "filter": ["lowercase"]
        }
      }
    }
  }
}
```
Put document: /tweets/tweet/1
```
{
  "text": "DEF-ABC"
}
```

Try to search: /_search
```
{
  "query": {
    "query_string": {
       "query": "*DEF*", "analyze_wildcard": true
    }
  }
}
```

No results, because org.elasticsearch.index.analysis.CustomAnalyzer doesn't apply token filters any more.

Prefix search: /_search
```
{
  "query": {
    "query_string": {
       "query": "DEF*", "analyze_wildcard": true
    }
  }
}
```

It returns a document as expected because org.apache.lucene.queryparser.classic.MapperQueryParser::getPossiblyAnalyzedPrefixQuery applies token filters during parsing of search terms.

I didn't test other queries but it breaks wildcard queries because it's not possible to apply any custom analyzer to search term for wildcard search term.

**Comment by andreybratukhin** [comment]
Also it ignores forced analyzer:
```
{
  "query": {
    "query_string": {
       "query": "*DEF*", "analyze_wildcard": true, "analyzer": "my_custom_lowercase"
    }
  }
}
```

**Comment by jimczi** [comment]
@andreybratukhin https://github.com/elastic/elasticsearch/pull/21919 has been merged to 5.2 only. 
5.1.1 does not have this change and as you described has a bug when a custom analyzer is used.
In this version the normalization of the wildcard term is not working when a custom analyzer is set.
This is fixed by https://github.com/elastic/elasticsearch/pull/21919 which applies the normalization to custom analyzer. It works with prefix query on 5.1.1 because we analyze the string (as opposed to only normalize) to find the last token on which the prefix should be applied. 
Bottom line is that ES 5.2 will not be affected by this bug, thanks for the report !

**Comment by andreybratukhin** [comment]
I haven't seen the version marker for the pull request. Thank you for clarification.

Do you have any suggestions for a workaround to 5.1.1 issue with custom analyzer and wildcards?

**Comment by jimczi** [comment]
Unfortunately not :(
Though you should be very careful with wildcards. Leading wildcard like in your example are very costly so my suggestion would be to forbid them ;) See `allow_leading_wildcard` in https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html ?
